### PR TITLE
DomainParticipant get_current_time

### PR DESCRIPTION
**Pull request by laura-eprosima**
* Add unit test for get_current_time functionality
* Implements current_time PSM method
* Add unit test for current_time PSM functionality

NOTE: This PR must be merged after #1073

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1631/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1397/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1627/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1645/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1411/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1641/)

File: include/dds/core/Time.hpp [hunk 1]
```diff
@@ -129,47 +148,161 @@ class OMG_DDS_API Time
...
+        }
+    }
 
     /**
      * @param that Time to compare
      * @return true if the Time is greater than or equal to the comparator
      */
     bool operator >=(
-            const Time& that) const;
+            const Time& that) const
```
Comment (IkerLuengo) [inline include/dds/core/Time.hpp]: Can be implemented as inline:
return !(*this < that)

It reduces code error chances keeping the logic in one place for all comparisons.

File: include/dds/core/Time.hpp [hunk 2]
```diff
@@ -129,47 +148,161 @@ class OMG_DDS_API Time
...
+        }
+    }
+
     /**
      * @param that Time to compare
      * @return true if the Time is less than the comparator
      */
     bool operator <(
-            const Time& that) const;
+            const Time& that) const
```
Comment (IkerLuengo) [inline include/dds/core/Time.hpp]: Can be implemented as inline:
return !(*this > that || *this == that)

It reduces code error chances keeping the logic in one place for all comparisons.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1688/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1449/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1680/)

**Pull request closed without merging**